src/main/java/pages/agoda/HomePage.java

```diff
@@ -0,0 +1,227 @@
+
+import static com.codeborne.selenide.Condition.visible;
+import static com.codeborne.selenide.Selenide.$;
+import static pages.agoda.HomePage.OccupancyType.ADULTS;
+import static pages.agoda.HomePage.OccupancyType.CHILDREN;
+import static pages.agoda.HomePage.OccupancyType.ROOMS;
+
+public class HomePage {
+
+    private final By googleIframe = By.cssSelector("iframe[src*='accounts.google.com']");
+    private final By iframeCloseBtn = By.id("close");
+    private final By menuIcon = By.className("HamburgerMenu");
+    private final By selectedCurrency = By.cssSelector("[data-element-name='currency-container-selected-currency-name']");
+    private final By vndCurrency = By.xpath("//p[text()='Vietnamese Dong']");
+    private final By searchBox = By.id("textInput");
+    private final By occContainer = By.className("OccupancySelector");
+    private final By searchBtn = By.cssSelector("[data-selenium='searchButton']");
+    private final By closeDownloadAppBtn = By.cssSelector("[leadingicon='fill.symbol.close']");
+
+    // dynamic locators
+    private final String searchSuggestion = "[data-text='%s']";
+    private final String selectedDate = "[data-selenium-date='%s']";
+
+
+    private int getRoomCount() {
+        return parseValueFromElement(OccupancyType.ROOMS.valueSelector());
+    }
+
+    private int getAdultCount() {
+        return parseValueFromElement(OccupancyType.ADULTS.valueSelector());
+    }
+
+    private int getChildCount() {
+        return parseValueFromElement(OccupancyType.CHILDREN.valueSelector());
+    }
+
+    private int parseValueFromElement(String selector) {
+        return Integer.parseInt($(selector).getText().trim());
+    }
+
+    // define an enum to manage occupancy's locators
+    @AllArgsConstructor
+    public enum OccupancyType {
+
+        ROOMS("occupancyRooms"),
+        ADULTS("occupancyAdults"),
+        CHILDREN("occupancyChildren");
+
+        private final String seleniumKey;
+
+        public String controlSelector() {
+            return "[data-selenium='%s']".formatted(seleniumKey);
+        }
+
+        public String plusButton() {
+            return controlSelector().concat(" [data-selenium='plus']");
+        }
+
+        public String minusButton() {
+            return controlSelector().concat(" [data-selenium='minus']");
+        }
+
+        public String valueSelector() {
+            return controlSelector().concat(" [data-component*='desktop-occ']");
+        }
+    }
+
+    @Step("Search a hotel with given information: {location}, {duration} days from next {targetDay}, occupancy: {occupancy}")
+    public void searchHotel(String location, DayOfWeek targetDay, int duration, Occupancy occupancy) {
+        selectVndCurrency();
+        searchForLocation(location);
+        selectDate(targetDay, duration);
+        setOccupancyTo(occupancy);
+        clickSearchButton();
+    }
+
+    @Step("Click on occupancy container")
+    public void selectOccupancyContainer() {
```

Comment:
Rename this method. Method names should reflect business steps rather than UI interactions.